File: queryUser.py
#!/usr/bin/python
from cassandra.cluster import Cluster
from cassandra.auth import PlainTextAuthProvider
import sys

hostname=sys.argv[1]
username=sys.argv[2]
password=sys.argv[3]
userid=sys.argv[4]

nodes = []
nodes.append(hostname)
keyspace="packt"

auth = PlainTextAuthProvider(username=username, password=password)
cluster = Cluster(nodes,auth_provider=auth)
session = cluster.connect(keyspace)

strINSERT = """
    INSERT INTO logins_by_user (user_id,login_datetime,origin_ip)
    VALUES (?,dateof(now()),?)
"""
pINSERTStatement = session.prepare(strINSERT);
session.execute(pINSERTStatement,['aploetz','192.168.0.114'])

strSELECT = """
    SELECT * FROM logins_by_user WHERE user_id=? LIMIT 3;
"""
pSELECTStatement = session.prepare(strSELECT);

rows = session.execute(pSELECTStatement,[userid])
print("Data for user %s:" % userid)
for row in rows:
    #only one row in system.local
    print(row[0] + " " +
          str(row[1]) + " " +
          row[2])

#closing Cassandra connection
session.shutdown()

File: cassandra/QueryUser.java
package CassHelloWorld;

import com.datastax.driver.core.BoundStatement;
import com.datastax.driver.core.PreparedStatement;
import com.datastax.driver.core.ResultSet;
import com.datastax.driver.core.Row;
import com.datastax.driver.core.Session;

public class QueryUser {

	public static void main(String[] args) {
		CassandraConnection conn = new CassandraConnection();
		conn.connect("192.168.0.100", "cassdba", "flynnLives");
		Session session = conn.getSession();

		String userID = System.getProperty("user.name");
		String strINSERT = "INSERT INTO packt.logins_by_user "
		    + "(user_id,login_datetime,origin_ip) "
		    + "VALUES (?,dateof(now()),?)";		

		PreparedStatement pIStatement = session.prepare(strINSERT);
		BoundStatement bIStatement = new BoundStatement(pIStatement);
		bIStatement.bind(userID, "192.168.0.119");
				
		String strSELECT = "SELECT * "
				+ "FROM packt.logins_by_user WHERE user_id=? "
				+ "LIMIT 3";
		PreparedStatement pSStatement = session.prepare(strSELECT);
		BoundStatement bSStatement = new BoundStatement(pSStatement);
		bSStatement.bind(userID);
		
		ResultSet rows = conn.query(bSStatement);
		for (Row row : rows) {
			System.out.println(row.getString("user_id") + " " +
					row.getTimestamp("login_datetime") + " " +
					row.getString("origin_ip"));
		}
		conn.close();
	}
}
